src/app/shared/lib/file-download.ts

```diff
@@ -0,0 +1,38 @@
+export function createAndDownloadFile(fileName: string, content: string) {
+    const blob: Blob = new Blob([content], {type: 'file/txt'});
+    const blobUrl: string = window.URL.createObjectURL(blob);
+
+    const hiddenElement: HTMLAnchorElement = createElement(blobUrl, fileName);
+
+    downloadElement(hiddenElement);
+
+    removeElement(blobUrl, hiddenElement);
+}
+
+function createElement(blobUrl: string, fileName: string): HTMLAnchorElement {
+    const hiddenElement: HTMLAnchorElement = document.createElement('a');
+    hiddenElement.setAttribute('style', 'display: none;');
+    hiddenElement.href = blobUrl;
+    hiddenElement.download = fileName;
+    return hiddenElement;
+}
+
+/**
+ * Appends HTML element to DOM and clicks to prompt download
+ * @param element: the file to be downloaded
+ */
+function downloadElement(element: HTMLAnchorElement) {
```

Comment:
```suggestion
function triggerDownload(element: HTMLAnchorElement) {
```

`downloadElement` doesn't sound right, since the element triggers the download

Reply:
Thank you for pointing that out! I will make the necessary changes and update the PR